src/lex/cpp.rs

```diff
@@ -726,6 +1062,28 @@ int f() BEGIN return 5; END";
         assert_same(opdef, "int f() { return 5; }");
     }
     #[test]
+    fn recursive_macros() {
```

Comment:
Not related to this test, but if you haven't already, you should add a redefinition test.